repos/fdbt-site/src/pages/products/productDetails.tsx

```diff
@@ -473,6 +478,35 @@ const createProductDetails = async (
         });
     }
 
+    if (ticket.type === 'flatFare' && 'pricingByDistance' in product) {
+        const pricePerDistance = product.pricingByDistance as DistancePricingData;
```

Comment:
Then once we have the typeguard, we wont need to cast the data like this

Reply:
pushed and removed the typecasting